### PR TITLE
Consistent naming + removes VCC CPU Version

### LINKED ISSUE #431: consistent naming scheme for traversals
**Is your feature request related to a problem? Please describe.**
currently, a user has little information, which traversal belongs to which container.

**Describe the solution you'd like**
This should be made clear by a consistent naming scheme.

```c++
// current, from TraversalOption.h:
        {TraversalOption::c08, "c08"},
        {TraversalOption::sliced, "sliced"},
        {TraversalOption::c18, "c18"},
        {TraversalOption::c01, "c01"},
        {TraversalOption::directSumTraversal, "directSum"},
        {TraversalOption::slicedVerlet, "verlet-sliced"},
        {TraversalOption::c18Verlet, "verlet-c18"},
        {TraversalOption::c01Verlet, "verlet-c01"},
        {TraversalOption::c01Cuda, "cuda-c01"},
        {TraversalOption::verletTraversal, "verlet-lists"},
        {TraversalOption::c01CombinedSoA, "c01-combined-SoA"},
        {TraversalOption::verletClusters, "verlet-clusters"},
        {TraversalOption::c04, "c04"},
        {TraversalOption::varVerletTraversalAsBuild, "var-verlet-lists-as-build"},
        {TraversalOption::verletClustersColoring, "verlet-clusters-coloring"},
        {TraversalOption::c04SoA, "c04SoA"},
        {TraversalOption::verletClusterCells, "verlet-cluster-cells"},
        {TraversalOption::verletClustersStatic, "verlet-clusters-static"},
```

proposed:
```c++
// proposed, TraversalOption.h:
        // LinkedCell Traversals:
        {TraversalOption::lc_sliced, "lc-sliced"},
        {TraversalOption::lc_c01, "lc-c01"},
        {TraversalOption::lc_c01_soa, "lc-c01-soa"},
        {TraversalOption::lc_c01_cuda, "lc-c01-cuda"},
        {TraversalOption::lc_c04, "lc-c04"},
        {TraversalOption::lc_c04_soa, "lc-c04-soa"},
        {TraversalOption::lc_c08, "lc-c08"},
        {TraversalOption::lc_c18, "lc-c18"},

        // DirectSum Traversals:
        {TraversalOption::ds_traversal, "ds-traversal"},

        // VerletList Traversals:
        {TraversalOption::vl_traversal, "vl-traversal"},

        // VARVL Traversals
        {TraversalOption::vl_asbuild, "vl-as-build"},

        // VerletListCells Traversals:
        {TraversalOption::vlc_sliced, "vlc-sliced"},
        {TraversalOption::vlc_c01, "vlc-c01"},
        {TraversalOption::vlc_c18, "vlc-c18"},

        // VerletClusterCells Traversals:
        {TraversalOption::vcc_traversal, "vcc-traversal"},

        // VerletClusterLists Traversals:
        {TraversalOption::vcl_traversal, "vcl-traversal"},
        {TraversalOption::vcl_coloring, "vcl-coloring"},
        {TraversalOption::vcl_static, "vcl-static"},
```

**Describe alternatives you've considered**
Open for suggestions.

**Additional context**
Also needs to be changed in the Value enum.

### PR DESCRIPTION
**Pull request by SteffenSeckler**
# Description

This makes the names of TraversalOptions consistent:

| old | new |
| --- | --- |
| directSumTraversal | ds_sequential |
| --- | --- |
| c01 | lc_c01 |
| c01CombinedSoA | lc_c01_combined_SoA |
| c01Cuda | lc_c01_cuda |
| c04 | lc_c04 |
| c04HCP | lc_c04_HCP |
| c04SoA | lc_c04_combined_SoA |
| c08 | lc_c08 |
| c18 | lc_c18 |
| sliced | lc_sliced |
| BalancedSliced | lc_sliced_balanced |
| cSliced | lc_sliced_c02 |
| --- | --- |
| verletClusters | vcl_cluster_iteration |
| verletClustersColoring | vcl_c06 |
| verletClustersStatic | vcl_c01_balanced |
| --- | --- |
| verletTraversal | vl_list_iteration |
| --- | --- |
| slicedVerlet | vlc_sliced |
| c18Verlet | vlc_c18 |
| c01Verlet | vlc_c01 |
| BalancedSlicedVerlet | vlc_sliced_balanced |
| cSlicedVerlet | vlc_sliced_c02 |
| --- | --- |
| verletClusterCells | vcc_cluster_iteration |
| --- | --- |
| varVerletTraversalAsBuild | vvl_as_built |

The string names of these are now also consistent and identical to the name of the enum value.

Also removes the CPU (AoS and SoA) versions of the VCC container.

## Resolved Issues

- [x] fixes #431 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [x] Fabio looked at it very hard !!!!

### CODE REVIEW

File: tests/testAutopas/tests/selectors/tuningStrategy/PredictiveTuningTest.h [hunk 1]
```diff
@@ -14,19 +15,69 @@
...
-      autopas::DataLayoutOption::soa, autopas::Newton3Option::disabled);
-  const autopas::Configuration configurationC08 = autopas::Configuration(
-      autopas::ContainerOption::linkedCells, 1., autopas::TraversalOption::c08, autopas::LoadEstimatorOption::none,
+  /**
+   * Tunes for a few iterations (length of times vector) and checks whether all possible configurations were tuned.
+   * @param predictiveTuning The PredictiveTuning strategy.
+   * @param evidences Array of times to add to the evidences.
+   * @param returnConfigIndices The configurations with these indices are returned.
+   * @param iteration The current number of iterations, will be increased accordingly.
+   * @return Array of the configurations with indices returnConfigIndices.
```
Comment (FG-TUM) [inline tests/testAutopas/tests/selectors/tuningStrategy/PredictiveTuningTest.h:24]:
```suggestion
   * Simulates tuning of a few iterations (given by the length of evidence vector) and checks whether all possible configurations were tuned.
   * @param predictiveTuning The PredictiveTuning strategy object.
   * @param evidence Array of times to add to the evidence.
   * @param returnConfigIndices The configurations with these indices are returned.
   * @param iteration Number of the first iteration that will be simulated.
   * @return Array of the configurations with indices returnConfigIndices.
```

Rename variable `evidences` to `evidence` ("evidence" is always singular)

returnConfigIndices is way to complicated. Just return everything.

File: tests/testAutopas/tests/selectors/tuningStrategy/FullSearchTest.cpp [hunk 2]
```diff
@@ -19,31 +19,31 @@ TEST_F(FullSearchTest, testSearchSpaceEmpty) {
 TEST_F(FullSearchTest, testTune) {
```
Comment (FG-TUM) [inline tests/testAutopas/tests/selectors/tuningStrategy/FullSearchTest.cpp:19]: shouldn't this be refactored similarly to the predictive tuning test?

Comment (SteffenSeckler) [reply tests/testAutopas/tests/selectors/tuningStrategy/FullSearchTest.cpp:19]: feel free ;)

Comment (SteffenSeckler) [reply tests/testAutopas/tests/selectors/tuningStrategy/FullSearchTest.cpp:19]: done myself

**Pull request merged**